src/Bicep.Core.Samples/Files/baselines/Resources_CRLF/main.json

```diff
@@ -629,6 +649,12 @@
       "apiVersion": "2021-11-01",
       "name": "sql-server-name",
       "location": "polandcentral"
+    },
+    {
+      "type": "Microsoft.Sql/servers",
+      "apiVersion": "2021-11-01",
+      "name": "[format('sql-server-nameof-{0}', '''server-name''')]",
```

Comment:
this doesn't seem right...

Reply:
The implementation for nameof on `ArrayAccessSyntax` will need to strip quotes on string keys, I think. 

Alternatively, it would be reasonable for nameof not to work on ArrayAccessSytanx nodes, which is how C# handles it:
![image](https://github.com/user-attachments/assets/9f576feb-1b86-4311-81b2-05bcc5d5ed25)
If the accessor expression is anything other than a quoted string, it's not obvious what `nameof` should return. For example, is `nameof(foo[fizz.buzz])` `buzz`, or does the expression raise an error? Same for `nameof(foo[0])`.

Reply:
I decided to allow literals as array access - it'll return the literal itself. any other usage returns error.